apps/api/src/user/service/user.service.ts

```diff
@@ -32,6 +39,32 @@ export class UserService {
       profilePictureUrl: dto?.profilePictureUrl,
       isOnboardingFinished: dto.isOnboardingFinished
     }
+    if (dto?.email) {
+      const userExists =
+        (await this.prisma.user.count({
+          where: {
+            email: dto.email
+          }
+        })) > 0
+
+      if (userExists) {
+        throw new ConflictException('User with this email already exists')
+      }
+
+      const otp = await generateOtp(user.email, user.id, this.prisma)
+
+      await this.prisma.userEmailChange.create({
```

Comment:
I think instead of just creating, we should use upsert. We will do a query by the `userId`. If it is found, we will update the value of `newEmail`. If not, we will create an entry.

You can get rid of `otp` and `expiresOn`

Replicate the same logic in `updateUser`. Alternatively, you can extract the email update code into a private function and pass in a flag called `isAdminTriggered`. Based on this, we will make the decision whether we want to send the otp or not.

Reply:
Done